actionpack/lib/action_controller/renderer.rb

```diff
@@ -103,8 +103,13 @@ def render(*args)
     private
       def normalize_keys(defaults, env)
         new_env = {}
-        defaults.each_pair { |k, v| new_env[rack_key_for(k)] = rack_value_for(k, v) }
         env.each_pair { |k, v| new_env[rack_key_for(k)] = rack_value_for(k, v) }
```

Comment:
Which is faster this vs `defaults.merge(env).each_pair { |k, v| new_env[rack_key_for(k)] = rack_value_for(k, v) }`?

```suggestion
        defaults.merge(env).each_pair { |k, v| new_env[rack_key_for(k)] = rack_value_for(k, v) }
```

Reply:
Since `defaults` and `env` has not compatible keys so `defaults.merge(env)` does not work as dedup keys.